src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -113,29 +114,57 @@ public static Classes parseClass(String courseCode) throws ParseException {
     }
 
     /**
-     * Parses a {@code String tag} into a {@code Tag}.
+     * Parses a {@code String date} into an {@code AttendanceDate}.
      * Leading and trailing whitespaces will be trimmed.
      *
-     * @throws ParseException if the given {@code tag} is invalid.
+     * @throws ParseException if the given {@code date} is invalid.
      */
-    public static Tag parseTag(String tag) throws ParseException {
-        requireNonNull(tag);
-        String trimmedTag = tag.trim();
-        if (!Tag.isValidTagName(trimmedTag)) {
-            throw new ParseException(Tag.MESSAGE_CONSTRAINTS);
+    public static Attendance parseDate(String date) throws ParseException {
+        requireNonNull(date);
+        String trimmedDate = date.trim();
+        if (!Attendance.isValidDate(trimmedDate)) {
+            throw new ParseException(Attendance.MESSAGE_CONSTRAINTS);
         }
-        return new Tag(trimmedTag);
+        return new Attendance(new AttendanceStatus(trimmedDate, "1"));
     }
 
     /**
-     * Parses {@code Collection<String> tags} into a {@code Set<Tag>}.
+     * Parses a {@code String date} into a {@code Attendance}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code attendance} is invalid.
+     */
+    public static Attendance parseAttendances(String date) throws ParseException {
+        requireNonNull(date);
+        String trimmedTag = date.trim();
+        if (!Attendance.isValidDate(trimmedTag)) {
+            throw new ParseException(Attendance.MESSAGE_CONSTRAINTS);
+        }
+        return new Attendance(new AttendanceStatus(trimmedTag, "1"));
+    }
+
+    /**
+     * Parses {@code Collection<String> attendances} into a {@code Set<Attendance>}.
+     */
+    public static Set<Attendance> parseAttendances(Collection<String> attendances) throws ParseException {
+        requireNonNull(attendances);
+        final Set<Attendance> attendanceSet = new HashSet<>();
+        for (String attendanceName : attendances) {
+            attendanceSet.add(parseAttendances(attendanceName));
+        }
+        return attendanceSet;
+    }
+
+    /**
+     * Parses {@code String date} and {@code String status} into a {@code AttendanceStatus}.
      */
-    public static Set<Tag> parseTags(Collection<String> tags) throws ParseException {
-        requireNonNull(tags);
-        final Set<Tag> tagSet = new HashSet<>();
-        for (String tagName : tags) {
-            tagSet.add(parseTag(tagName));
+    public static AttendanceStatus parsesAttendanceStatus(String date, String status) throws ParseException {
+        requireNonNull(date, status);
+        String trimmedDate = date.trim();
+        String trimmedStatus = status.trim();
+        if (!Attendance.isValidDate(trimmedDate) || !Attendance.isValidStatus(trimmedStatus)) {
```

Comment:
is it better to make ```!Attendance.isValidDate(trimmedDate)``` and ```!Attendance.isValidStatus(trimmedStatus)``` separate conditions to make it clearer when there is error thrown?